### ISSUE TITLE
JSON objects are treated like collections instead of objects with Assert.Equivalent

### ISSUE THREAD

**Issue by piero-b** [xunit/xunit#2716]
Hi, when trying to test if two JsonNodes are the same this results in an exception being thrown "An item with the same key has already been added." If one of the keys are changed to a different name, then the assertion runs to show the failure.

```
[Fact]
public void Parse_json_string_into_json_node()
{
	Assert.Equivalent(JsonNode.Parse("{\"some_key\":\"some_value\"}"), JsonNode.Parse("{\"some_key\":\"some_value\"}"));
}
```

```
System.ArgumentException: An item with the same key has already been added. Key: Item
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
   at Xunit.Internal.AssertHelper.<>c.<GetGettersForType>b__1_0(Type _type) in /_/src/xunit.assert/Asserts/Sdk/AssertHelper.cs:line 53
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Xunit.Internal.AssertHelper.GetGettersForType(Type type) in /_/src/xunit.assert/Asserts/Sdk/AssertHelper.cs:line 31
   at Xunit.Internal.AssertHelper.VerifyEquivalenceReference(Object expected, Object actual, Boolean strict, String prefix, HashSet`1 expectedRefs, HashSet`1 actualRefs) in /_/src/xunit.assert/Asserts/Sdk/AssertHelper.cs:line 227
   at Xunit.Internal.AssertHelper.VerifyEquivalence(Object expected, Object actual, Boolean strict, String prefix, HashSet`1 expectedRefs, HashSet`1 actualRefs) in /_/src/xunit.assert/Asserts/Sdk/AssertHelper.cs:line 165
   at Xunit.Internal.AssertHelper.VerifyEquivalence(Object expected, Object actual, Boolean strict, String prefix, HashSet`1 expectedRefs, HashSet`1 actualRefs) in /_/src/xunit.assert/Asserts/Sdk/AssertHelper.cs:line 144
   at Xunit.Internal.AssertHelper.VerifyEquivalenceEnumerable(IEnumerable expected, IEnumerable actual, Boolean strict, String prefix, HashSet`1 expectedRefs, HashSet`1 actualRefs) in /_/src/xunit.assert/Asserts/Sdk/AssertHelper.cs:line 200
   at Xunit.Internal.AssertHelper.VerifyEquivalence(Object expected, Object actual, Boolean strict, String prefix, HashSet`1 expectedRefs, HashSet`1 actualRefs) in /_/src/xunit.assert/Asserts/Sdk/AssertHelper.cs:line 163
   at Xunit.Internal.AssertHelper.VerifyEquivalence(Object expected, Object actual, Boolean strict) in /_/src/xunit.assert/Asserts/Sdk/AssertHelper.cs:line 71
   at Xunit.Assert.Equivalent(Object expected, Object actual, Boolean strict) in /_/src/xunit.assert/Asserts/EquivalenceAsserts.cs:line 39
```

**Comment by bradwilson** [comment]
The throw is already fixed in `2.5.0-pre.11`.

That said, the test fails:

```
Assert.Equivalent() Failure: Collection value not found
Expected: ["some_key"] = some_value
In:       [["some_key"] = some_value]
```

That seems as though it's because your `JsonNode` is being treated like a collection, whereas it _feels_ like it should be treated like a raw object. I'm wondering if there's some way for us to do anything slightly more intelligent here... So I'm going to convert this issue to that.

**Comment by lateapexearlyspeed** [comment]
Another way to do JSON assertion is to just use [JSON assertion library](https://github.com/lateapexearlyspeed/Lateapexearlyspeed.JsonSchema?tab=readme-ov-file#lateapexearlyspeedxunitassertionjson) which doesn't depend on object types to do equivalent test and can report failed json node location :)
